airbyte-integrations/connectors/source-github/unit_tests/integration/test_events.py

```diff
@@ -0,0 +1,158 @@
+                            "graphql": {"limit": 5000, "used": 0, "remaining": 5000, "reset": 5070908800},
+                        }
+                    }
+                ),
+                200,
+            ),
+        )
+
+        self.r_mock.get(
+            HttpRequest(
+                url=f"https://api.github.com/repos/{_CONFIG.get('repositories')[0]}",
+                query_params={"per_page": 100},
+            ),
+            HttpResponse(json.dumps({"full_name": "airbytehq/integration-test", "default_branch": "master"}), 200),
+        )
+
+        self.r_mock.get(
+            HttpRequest(
+                url=f"https://api.github.com/repos/{_CONFIG.get('repositories')[0]}/branches",
+                query_params={"per_page": 100},
+            ),
+            HttpResponse(json.dumps([{"repository": "airbytehq/integration-test", "name": "master"}]), 200),
+        )
+
+    def teardown(self):
+        """Stops and resets HttpMocker instance."""
+        self.r_mock.__exit__()
+
+    def test_full_refresh_no_pagination(self):
+        """Ensure http integration, record extraction and transformation"""
+
+        self.r_mock.get(
+            HttpRequest(
+                url=f"https://api.github.com/repos/{_CONFIG.get('repositories')[0]}/events",
+                query_params={"per_page": 100},
+            ),
+            HttpResponse(json.dumps(find_template("events", __file__)), 200),
+        )
+
+        source = SourceGithub()
+        actual_messages = read(source, config=_CONFIG, catalog=_create_catalog())
+
+        assert len(actual_messages.records) == 2
+        assert all(("repository", "airbytehq/integration-test") in x.record.data.items() for x in actual_messages.records)
+
+    def test_full_refresh_with_pagination(self):
+        """Ensure pagination"""
+        self.r_mock.get(
+            HttpRequest(
+                url=f"https://api.github.com/repos/{_CONFIG.get('repositories')[0]}/events",
+                query_params={"per_page": 100},
+            ),
+            HttpResponse(
+                body=json.dumps(find_template("events", __file__)),
+                status_code=200,
+                headers={"Link": '<https://api.github.com/repos/{}/events?page=2>; rel="next"'.format(_CONFIG.get("repositories")[0])},
+            ),
+        )
+        self.r_mock.get(
+            HttpRequest(
+                url=f"https://api.github.com/repos/{_CONFIG.get('repositories')[0]}/events",
+                query_params={"per_page": 100, "page": 2},
+            ),
+            HttpResponse(
+                body=json.dumps(find_template("events", __file__)),
+                status_code=200,
+            ),
+        )
+        source = SourceGithub()
+        actual_messages = read(source, config=_CONFIG, catalog=_create_catalog())
+
+        assert len(actual_messages.records) == 4
+
+    def test_incremental_read(self):
```

Comment:
Could we have a more descriptive name? Something like `test_given_state_more_recent_than_some_records_when_read_incrementally_then_filter_records`

Reply:
renamed